core/src/main/java/io/questdb/cutlass/http/processors/QueryCache.java

```diff
@@ -41,10 +45,13 @@ public final class QueryCache implements Closeable {
     private static ThreadLocal<QueryCache> TL_QUERY_CACHE;
     private static HttpServerConfiguration httpServerConfiguration;
     private static Metrics metrics;
-    private final AssociativeCache<RecordCursorFactory> cache;
+    private final AssociativeCache<FactoryAndPermissions> cache;
+
+    private final WeakSelfReturningObjectPool<FactoryAndPermissions> factoryAndPermissionsPool;
```

Comment:
can see only `pop()` from this pool, and no `push()` ever.
is that expected?
if there is only `pop()` do we need the pool at all, maybe we can just use `new` instead?
or maybe in QueryCache::clear() the objects can be added back into the pool?

Reply:
FactoryAndPermissions  objects should go back to the pool on close() .

Reply:
got it.
AbstractSelfReturningObject::close() calls the push().